posts/bayesian-model-comparison/index.qmd

```diff
@@ -0,0 +1,470 @@
+---
+title: Bayesian comparison of cross-validated algorithms
+subtitle: 'Model selection via a Bayesian correlated t-test.'
+date: 2022/11/11
+image: thumb.png
+categories:
+  - ML
+  - Theory
+  - Model evaluation
+  - Bayes
+  - Bayesian estimation
+  - Cross-validation
+bibliography: ../references.bib
+jupyter:
+  jupytext:
+    text_representation:
+      extension: .qmd
+      format_name: quarto
+      format_version: '1.0'
+      jupytext_version: 1.13.7
+  kernelspec:
+    display_name: Python 3.6.15 ('fgm')
+    language: python
+    name: python3
+---
+
+```{python}
+#| eval: false
+#| echo: false
+!pip install -r requirements.txt
+```
+
+Robust model evaluation should be second nature for those of us that work with data and 
+predictive models. 
+When determining whether one model is better than another, there are many techniques one can use.
+Very common ones include *bootstrapping* and *cross-validation*. 
+
+First and foremost however, one must make a decision as to what "better" means.
+Better in what way? 
+More accurate? 
+Faster? 
+More (budget) efficient? 
+We will come to this point again later.
+For now, let's assume that better means "obtains a higher F1 score".
+F1 is just an example, and we could consider any other score. 
+Once we have decided to say that the model with highest F1 score is better, the more serious question arises: how do we know whether a model is indeed better? 
+Can we just take an average score across a test set and compare the numbers, i.e. plain-old pointwise comparison? 
+I think many will get a sense of discomfort just uttering that. 
+We would like something more robust and which, ideally, takes uncertainty into account. 
+
+We ideally want to have a probability distribution which informs us, in some way, what the probability of one model being better than the other is. 
+That is, we want something like $P(M_1 > M_2 | D)$, where $M_1 > M_2$ denotes that model $M_1$ is better than model $M_2$ and $D$ is the data used for the comparison.
+In fact, we may want to do more than that. 
+Ideally, we would want a distribution specifying what the probability of one model being better than another is for each given performance difference. 
+That is, we would like a distribution over the true performance difference of the two algorithms.
```

Comment:
This still confuses me a little. Sorry being very slow. How about, simply:

> Ideally, we would want a distribution over the true performance difference of the two algorithms.

Reply:
Yes, much better actually. I'll change

- [ ] change description

Reply:
Ha, just realised I'd written exactly that below. had misunderstood your comment. Will delete the "duplication"